back-end/yelpCrawler.py

```diff
@@ -0,0 +1,203 @@
+def search(api_key, term, location, offset):
+    """Query the Search API by a search term and location.
+    Args:
+        term (str): The search term passed to the API.
+        location (str): The search location passed to the API.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    url_params = {
+        'term': term.replace(' ', '+'),
+        'location': location.replace(' ', '+'),
+        #'latitude': latitude,
+        #'longitude': longitude,
+        'open_now': OPEN_NOW.replace(' ', '+'),
+        'limit': SEARCH_LIMIT,
+        'offset': offset
+    }
+    return request(API_HOST, SEARCH_PATH, api_key, url_params=url_params)
+
+
+def get_business(api_key, business_id):
+    """Query the Business API by a business ID.
+    Args:
+        business_id (str): The ID of the business to query.
+    Returns:
+        dict: The JSON response from the request.
+    """
+    business_path = BUSINESS_PATH + business_id
+
+    return request(API_HOST, business_path, api_key)
+
+# Define default place : wework civic, default wifi='free', default parking='street'
+# def getYelpData(lat=37.778264, longi=-122.411843, wifi="Free", parking="Street"):
+def getYelpData(wifi="Free", parking="Street"):
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-s', '--term', dest='term', default=DEFAULT_TERM,
+                        type=str, help='Search term (default: %(default)s)')
+    parser.add_argument('-l', '--location', dest='location', default=DEFAULT_LOCATION,
+                        type=str, help='Search term (default: %(default)s)')
+    #parser.add_argument('-la', '--longitude', dest='longitude', default=LONGITUDE,
+    #                    type=float, help='Search longitude (default: %(default)f)')
+    #parser.add_argument('-lo', '--latitude', dest='latitude', default=LATITUDE,
+    #                    type=float, help='Search latitude (default: %(default)f)')
+
+    input_values = parser.parse_args()
+    total_data_list = []
+    search_limit = 50
+
+    for i in range(20):
+        try:
+            #response = search(API_KEY, input_values.term, lat, longi, search_limit*i)
+            response = search(API_KEY, input_values.term, input_values.location, 1 + (search_limit * i))
+            businesses = response.get('businesses')
+            businessList = []
+            for idx in businesses:
+                response = get_business(API_KEY, idx['id'])
+                businessList.append(response)
+        except HTTPError as error:
+            sys.exit(
+                'Encountered HTTP error {0} on {1}:\n {2}\nAbort program.'.format(
+                    error.code,
+                    error.url,
+                    error.read(),
+                )
+            )
+
+        # Merge loadJsonFile.py in this file
+        datainfo = businessList
+
+        datalist = []
+        for item in datainfo:
+            data = dict()
+            if(item.get('name')): data['name'] = item['name']
```

Comment:
Usually it's not recommended to repeatedly use hardcoded string (too many 'name', 'location', and so on). Can you introduce a variable that holds the string, and use that?